fix: bugs when serialize union type (#1655)
docarray/docarray: docs/user_guide/sending/serialization.md

assert isinstance(dv_from_proto_numpy.tensor, NdArray)
```

!!! note

>     Currently, union type hints are not supported for serialization to protobuf formats.

Reviewer: Any union type? or Union types of regular types are? Can u add tests for the exceptions?

Author: I'll add more tests. It should be any union type as union type doesn't have the `from_protobuf` method

assert isinstance(dv_from_proto_numpy.tensor, NdArray)
```

!!! note

>     Serialization to protobuf formats is not supported for union type hints (unless your union type consists solely of basic types). Please use JSON serialization as an alternative for these cases.

Reviewer: ```suggestion
    Serialization to protobuf is not supported for union types involving `BaseDoc` types.
```

```

!!! note

>     Serialization to protobuf formats is not supported for union type hints. Please consider using JSON serialization as an alternative.

Reviewer: Avoid the consideration, just inform
